tests/build-test-report.js

```diff
@@ -254,6 +248,68 @@ function runTests(outputFilePath, verboseOutputFilePath) {
   });
 }
 
+function downloadCircleResults() {
+  console.log("downloading test results from circle ci...");
+  const VCS_TYPE = 'github';
+  const USERNAME = 'code-dot-org';
+  const PROJECT = 'JS-Interpreter';
+  const REQUEST_PATH = `https://circleci.com/api/v1.1/project/${VCS_TYPE}/${USERNAME}/${PROJECT}/${argv.circleBuild}/artifacts`;
+
+
+  return new Promise((resolve, reject) => request(
+    {
+      url: REQUEST_PATH,
+      json: true,
+    },
+    (error, response, artifacts) => {
+      if (error) {
+        throw error;
+      }
+
+      const resultFileUrls = artifacts
+        .filter(a => a.pretty_path === '$CIRCLE_ARTIFACTS/test-results-new.json')
+        .map(a => a.url);
+
+      const bar = new ProgressBar(
+        '[:bar] :current/:total',
+        {
+          curr: 0,
+          total: artifacts.length,
+        });
+
+      Promise
+        .all(
+          resultFileUrls.map(
```

Comment:
Nit: I actually feel like _less_ whitespace would be more readable for the next ten lines or so.  So many extra layers of indentation here!  What's wrong with

```javascript
Promise.all(resultFileUrls.map(url =>
  new Promise((resolve, reject) =>
    request({url: url, json: true}, (error, _, body) => {
```